diff --git a/examples/platform/nxp/rt/rt1060/app/project_include/FreeRTOSConfig.h b/examples/platform/nxp/rt/rt1060/app/project_include/FreeRTOSConfig.h
index 3972a4555e..32cdc25966 100644
--- a/examples/platform/nxp/rt/rt1060/app/project_include/FreeRTOSConfig.h
+++ b/examples/platform/nxp/rt/rt1060/app/project_include/FreeRTOSConfig.h
@@ -42,7 +42,7 @@
 
 #define configUSE_PREEMPTION                         1
 #define configUSE_IDLE_HOOK                          1
-#define configUSE_TICK_HOOK                          CHIP_DEVICE_CONFIG_ENABLE_WPA
+#define configUSE_TICK_HOOK                          CHIP_DEVICE_CONFIG_ENABLE_WPA || CHIP_DEVICE_CONFIG_ENABLE_DISPLAY
 #define configUSE_TICKLESS_IDLE                      0
 #define configUSE_DAEMON_TASK_STARTUP_HOOK           0
 #define configCPU_CLOCK_HZ                           ( SystemCoreClock )
@@ -52,7 +52,7 @@
 /* stack size increased for NVM/LITTLE_FS save in idle task */
 #define configMINIMAL_STACK_SIZE                     ( ( uint16_t ) 2048  )
 #ifndef configTOTAL_HEAP_SIZE
-#define configTOTAL_HEAP_SIZE                        ( ( size_t ) ( 120 * 1024 ) )
+#define configTOTAL_HEAP_SIZE                        ( ( size_t ) ( 122 * 1024 ) )
 #endif
 #define configAPPLICATION_ALLOCATED_HEAP             1
 #define configSUPPORT_STATIC_ALLOCATION              0
