diff --git a/src/platform/nxp/rt/rt1060/PlatformManagerImpl.cpp b/src/platform/nxp/rt/rt1060/PlatformManagerImpl.cpp
index 97bbc1a3da..353ef283fc 100644
--- a/src/platform/nxp/rt/rt1060/PlatformManagerImpl.cpp
+++ b/src/platform/nxp/rt/rt1060/PlatformManagerImpl.cpp
@@ -458,6 +458,11 @@ bool PlatformManagerImpl::GetResetInIdleValue(void)
     return resetInIdle;
 }
 
+extern "C" bool getResetInIdleValue(void)
+{
+    return PlatformMgrImpl().GetResetInIdleValue();
+}
+
 void PlatformManagerImpl::StopBLEConnectivity(void)
 {
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
